parsl/monitoring/monitoring.py

```diff
@@ -559,17 +576,10 @@ def monitor(pid,
             d['psutil_process_time_system'] += total_children_system_time
             logging.debug("sending message")
             radio.send(d)
-            first_msg = False
         except Exception:
             logging.exception("Exception getting the resource usage. Not sending usage to Hub", exc_info=True)
 
-        try:
-            msg = command_q.get(block=False)
-            if msg == "Finished":
-                logging.info("Received task finished message. Ending the monitoring loop now.")
-                break
-        except queue.Empty:
-            logging.debug("Have not received any message.")
-
         logging.debug("sleeping")
         time.sleep(sleep_dur)
+
+    logger.info("Monitor exiting")
```

Comment:
This statement is unreachable (and so upsets  `mypy` in the benc-mypy branch). This process doesn't exit by reaching the end of its main function; instead it is going to exit by being terminated from outside by the monitor wrapper process. Putting this statement in a `finally` block might cause it to run in that case, though.